dht-node/src/dht_node/index.ts

```diff
@@ -0,0 +1,187 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+import DHT from '@hyperswarm/dht'
+import { logger } from '@/server/logger'
+import CONFIG from '@/config'
+import { Community as DbCommunity } from '@entity/Community'
+
+const KEY_SECRET_SEEDBYTES = 32
+const getSeed = (): Buffer | null =>
+  CONFIG.FEDERATION_DHT_SEED ? Buffer.alloc(KEY_SECRET_SEEDBYTES, CONFIG.FEDERATION_DHT_SEED) : null
+
+const POLLTIME = 20000
+const SUCCESSTIME = 120000
+const ERRORTIME = 240000
+const ANNOUNCETIME = 30000
+
+enum ApiVersionType {
+  V1_0 = 'v1_0',
+  V1_1 = 'v1_1',
+  V2_0 = 'v2_0',
+}
+type CommunityApi = {
+  api: string
+  url: string
+}
+
+export const startDHT = async (topic: string): Promise<void> => {
+  try {
+    const TOPIC = DHT.hash(Buffer.from(topic))
+    const keyPair = DHT.keyPair(getSeed())
+    logger.info(`keyPairDHT: publicKey=${keyPair.publicKey.toString('hex')}`)
+    logger.debug(`keyPairDHT: secretKey=${keyPair.secretKey.toString('hex')}`)
+
+    const ownApiVersions = Object.values(ApiVersionType).map(function (apiEnum) {
+      const comApi: CommunityApi = {
+        api: apiEnum,
+        url: CONFIG.FEDERATION_COMMUNITY_URL + apiEnum,
+      }
+      return comApi
+    })
+    logger.debug(`ApiList: ${JSON.stringify(ownApiVersions)}`)
+
+    const node = new DHT({ keyPair })
+
+    const server = node.createServer()
+
+    server.on('connection', function (socket: any) {
+      logger.info(`server on... with Remote public key: ${socket.remotePublicKey.toString('hex')}`)
+
+      socket.on('data', async (data: Buffer) => {
+        try {
+          if (data.length > 1141) {
+            logger.warn(
+              `received more than max allowed length of data buffer: ${data.length} against 1141 max allowed`,
+            )
+            return
+          }
+          logger.info(`data: ${data.toString('ascii')}`)
+          const recApiVersions: CommunityApi[] = JSON.parse(data.toString('ascii'))
+
+          // TODO better to introduce the validation by https://github.com/typestack/class-validato
+          if (recApiVersions && Array.isArray(recApiVersions) && recApiVersions.length < 5) {
```

Comment:
Why change the followed logic in this case? Please test the negative case and log and return.
```suggestion
          if (!recApiVersions && !Array.isArray(recApiVersions) && recApiVersions.length > 4) {
            logger.warn(
              `received totaly wrong or too much apiVersions-Definition JSON-String: ${JSON.stringify(
                recApiVersions,
              )}`,
            )
            return
          }
```

Reply:
fixes. Btw you also have to change `&&` to `||`. ;)

Reply:
Good catch!